assets/js/custom/MyProfile.js

```diff
@@ -135,18 +135,66 @@ let MyProfile = function (profile_url, save_username,
     $.get(self.toggle_visibility_url + '/' + id, {}, function (data) {
       let visibilityLockId = $('#visibility-lock-' + id)
       let visibilityLockOpenId = $('#visibility-lock-open-' + id)
+      let programName = $('#program-' + id).find('.program-name').text()
+  
+      let translations = []
+      translations.push({
+        key  : '%programName%',
+        value: programName
+      })
+     
+      
+      let url = Routing.generate('translate_word', {
+        'word'  : 'programs.changeVisibility',
+        'array' : JSON.stringify(translations),
+        'domain': 'catroweb'
+      }, false)
+      
       if (data === 'true')
       {
+  
+        $.get(url,function(data){
+          let split = data.split('\n')
         if (visibilityLockId.is(':visible'))
         {
-          visibilityLockId.hide()
-          visibilityLockOpenId.show()
+        
+  
+  
+            swal({
+            title             : split[0],
+            html              : split[3],
+            type              : 'warning',
+            showCancelButton  : true,
+            confirmButtonColor: '#3085d6',
+            cancelButtonColor : '#d33',
+            confirmButtonText : split[4],
+            cancelButtonText  : split[5],
+          }).then(() => {
+            visibilityLockId.hide()
+            visibilityLockOpenId.show()
+          }).catch(swal.noop)
+        
+          
+          
         }
         else
         {
-          visibilityLockOpenId.hide()
-          visibilityLockId.show()
+          swal({
+            title: split[0],
+            html:  split[1] + '<br><br>' + split[2],
+            type:   'warning',
+            showCancelButton  : true,
+            confirmButtonColor: '#3085d6',
```

Comment:
I guess it would be nice if we change the hardcoded values in this file to variables (confirm and cancel button color)

The duplicated code of swals per se is not really nice, but I would say that should be a separate ticket.
Therefore I would say you can leave it that way for now